File: wallet/src/main/java/com/worldventures/wallet/analytics/locatecard/action/BaseLocateSmartCardAction.java
package com.worldventures.wallet.analytics.locatecard.action;

import com.worldventures.core.service.analytics.Attribute;
import com.worldventures.wallet.analytics.WalletAnalyticsAction;
import com.worldventures.wallet.domain.entity.lostcard.WalletCoordinates;

import java.util.Locale;

public abstract class BaseLocateSmartCardAction extends WalletAnalyticsAction {

   @Attribute("coordinates") String coordinates;

   public void setLocation(WalletCoordinates walletCoordinates) {
      this.coordinates = String.format(Locale.US, "%f,%f", walletCoordinates.lat(), walletCoordinates.lng());
   }
}

File: wallet/src/main/java/com/worldventures/wallet/analytics/locatecard/action/BaseLocateSmartCardAction.kt
package com.worldventures.wallet.analytics.locatecard.action

import com.worldventures.core.service.analytics.Attribute
import com.worldventures.wallet.analytics.WalletAnalyticsAction
import com.worldventures.wallet.domain.entity.lostcard.WalletCoordinates

abstract class BaseLocateSmartCardAction : WalletAnalyticsAction() {

   @Attribute("coordinates") internal var coordinates: String? = null

   open fun setLocation(walletCoordinates: WalletCoordinates) {
      this.coordinates = "${walletCoordinates.lat},${walletCoordinates.lng}"
   }
}
